beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -115,6 +104,48 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		}
 	}
 
+	if !s.cfg.chain.InForkchoice(bytesutil.ToBytes32(data.BeaconBlockRoot)) {
+		tracing.AnnotateError(span, blockchain.ErrNotDescendantOfFinalized)
+		return pubsub.ValidationIgnore, blockchain.ErrNotDescendantOfFinalized
+	}
+
+	if err = s.cfg.chain.VerifyLmdFfgConsistency(ctx, att); err != nil {
```

Comment:
Same here and for the rest of the methods that follow it. If it is difficult enough, you should just stream the attestations after